core/src/main/scala/org/apache/spark/util/ShutdownHookManager.scala

```diff
@@ -177,8 +181,19 @@ private [util] class SparkShutdownHookManager {
     val hookTask = new Runnable() {
       override def run(): Unit = runAll()
     }
-    org.apache.hadoop.util.ShutdownHookManager.get().addShutdownHook(
-      hookTask, FileSystem.SHUTDOWN_HOOK_PRIORITY + 30)
+    val priority = FileSystem.SHUTDOWN_HOOK_PRIORITY + 30
+    // The timeout property must be passed as a Java system property because this
```

Comment:
According to this claim, why don't we simply use `System.getProperty` instead of creating `SparkConf`? Then, we don't need to introduce this `SparkConf` dependency to this `ShutdownHookManager` class.

Reply:
The reason I chose to use SparkConf is because it comes with parsing the value without extra code. That includes support for different time units, converting to milliseconds, and handling the optional case.